core/src/main/java/org/apache/iceberg/PartitionStatsHandler.java

```diff
@@ -226,7 +278,12 @@ public static CloseableIterable<PartitionStats> readPartitionStatsFile(
 
     CloseableIterable<StructLike> records =
         InternalData.read(fileFormat, inputFile).project(schema).build();
-    return CloseableIterable.transform(records, PartitionStatsHandler::recordToPartitionStats);
+
+    if (schema.findField(DV_COUNT.name()) == null) {
+      return CloseableIterable.transform(records, PartitionStatsHandler::recordToPartitionStats);
+    } else {
+      return CloseableIterable.transform(records, PartitionStatsHandler::recordToPartitionStatsV3);
```

Comment:
added a new function `recordToPartitionStatsV3` instead of passing the format version to `recordToPartitionStats` because `CloseableIterable.transform` needs a `Function` interface (which accepts one argument)